### ISSUE TITLE
Low Accuracy in Image Classification Problem

### ISSUE THREAD

**Issue by TheAwerx** [dotnet/machinelearning-modelbuilder#2163]
**System Information (please complete the following information):**
 - Model Builder Version (available in Manage Extensions dialog): 16.13.6.2226201
 - Visual Studio Version: Microsoft Visual Studio Community 2022 (64-bit) - Current
Version 17.2.3

**Describe the bug**
- On which step of the process did you run into an issue: Training
- Clear description of the problem: I have tried different datasets but the accuracy is everytime under 0,6. My dataset contains 150 classes and 500 images in each class.
- Total epochs is 23 which is very low.
-  My console looks like this after training is done:
|     Trainer                              MicroAccuracy  MacroAccuracy  Duration #Iteration                     |
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:   0, Accuracy:  0,2555556, Cross-Entropy:   3,372909, Learning Rate:       0,01
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:   0, Accuracy:  0,3703333, Cross-Entropy:   2,665637
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:   1, Accuracy:  0,4627776, Cross-Entropy:   2,231869, Learning Rate:       0,01
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:   1, Accuracy:  0,4193333, Cross-Entropy:   2,398289
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:   2, Accuracy:  0,5498894, Cross-Entropy:   1,861059, Learning Rate:     0,0094
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:   2, Accuracy:  0,4473333, Cross-Entropy:   2,297006
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:   3, Accuracy:  0,6014449, Cross-Entropy:   1,649867, Learning Rate:     0,0094
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:   3, Accuracy:  0,4703334, Cross-Entropy:   2,252339
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:   4, Accuracy:  0,6430001, Cross-Entropy:   1,484302, Learning Rate:   0,008836
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:   4, Accuracy:  0,4733334, Cross-Entropy:   2,224294
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:   5, Accuracy:  0,6776665, Cross-Entropy:    1,36826, Learning Rate:   0,008836
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:   5, Accuracy:  0,4763333, Cross-Entropy:   2,213967
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:   6, Accuracy:  0,7056668, Cross-Entropy:   1,262957, Learning Rate: 0,008305839
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:   6, Accuracy:  0,4743333, Cross-Entropy:   2,204843
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:   7, Accuracy:   0,725445, Cross-Entropy:    1,18535, Learning Rate: 0,008305839
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:   7, Accuracy:  0,4713333, Cross-Entropy:   2,205068
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:   8, Accuracy:  0,7484449, Cross-Entropy:   1,109881, Learning Rate: 0,007807489
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:   8, Accuracy:  0,4653334, Cross-Entropy:   2,202923
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:   9, Accuracy:   0,763667, Cross-Entropy:   1,053175, Learning Rate: 0,007807489
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:   9, Accuracy:  0,4683334, Cross-Entropy:   2,207179
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:  10, Accuracy:  0,7784448, Cross-Entropy:  0,9956066, Learning Rate: 0,00733904
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:  10, Accuracy:  0,4713334, Cross-Entropy:    2,20806
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:  11, Accuracy:  0,7903339, Cross-Entropy:  0,9519703, Learning Rate: 0,00733904
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:  11, Accuracy:  0,4713333, Cross-Entropy:   2,213974
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:  12, Accuracy:  0,8027785, Cross-Entropy:  0,9063346, Learning Rate: 0,006898697
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:  12, Accuracy:  0,4733333, Cross-Entropy:   2,216309
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:  13, Accuracy:  0,8135563, Cross-Entropy:  0,8715954, Learning Rate: 0,006898697
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:  13, Accuracy:  0,4713334, Cross-Entropy:   2,222885
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:  14, Accuracy:  0,8261118, Cross-Entropy:   0,834453, Learning Rate: 0,006484775
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:  14, Accuracy:  0,4693334, Cross-Entropy:    2,22594
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:  15, Accuracy:  0,8346676, Cross-Entropy:  0,8061182, Learning Rate: 0,006484775
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:  15, Accuracy:  0,4683333, Cross-Entropy:   2,232685
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:  16, Accuracy:  0,8436677, Cross-Entropy:  0,7752995, Learning Rate: 0,006095689
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:  16, Accuracy:  0,4673333, Cross-Entropy:   2,236058
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:  17, Accuracy:  0,8507786, Cross-Entropy:  0,7517623, Learning Rate: 0,006095689
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:  17, Accuracy:  0,4643334, Cross-Entropy:   2,242702
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:  18, Accuracy:  0,8591117, Cross-Entropy:  0,7258064, Learning Rate: 0,005729948
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:  18, Accuracy:  0,4663334, Cross-Entropy:   2,246158
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:  19, Accuracy:  0,8652231, Cross-Entropy:  0,7059726, Learning Rate: 0,005729948
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:  19, Accuracy:  0,4663334, Cross-Entropy:   2,252548
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:  20, Accuracy:  0,8713344, Cross-Entropy:  0,6838518, Learning Rate: 0,005386151
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:  20, Accuracy:  0,4683333, Cross-Entropy:   2,255949
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:  21, Accuracy:  0,8751123, Cross-Entropy:  0,6669409, Learning Rate: 0,005386151
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:  21, Accuracy:  0,4693333, Cross-Entropy:   2,262003
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:  22, Accuracy:  0,8812236, Cross-Entropy:  0,6479013, Learning Rate: 0,005062982
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:  22, Accuracy:  0,4653334, Cross-Entropy:   2,265272
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used:      Train, Batch Processed Count: 900, Epoch:  23, Accuracy:  0,8871126, Cross-Entropy:   0,633343, Learning Rate: 0,005062982
[Source=ImageClassificationTrainer; ImageClassificationTrainer, Kind=Trace] Phase: Training, Dataset used: Validation, Batch Processed Count: 100, Epoch:  23, Accuracy:  0,4653334, Cross-Entropy:   2,270955
|0    DNN + ResNet50                              0,4610         0,4624     228,2          0                     |

===============================================Experiment Results=================================================
------------------------------------------------------------------------------------------------------------------
|                                                     Summary                                                    |
------------------------------------------------------------------------------------------------------------------
|ML Task: ImageClassification                                                                                    |
|Dataset:                                                                                                        |
|Label : Label                                                                                                   |
|Total experiment time : 228,21 Secs                                                                             |
|Total number of models explored: 1                                                                              |
------------------------------------------------------------------------------------------------------------------

|                                              Top 1 models explored                                             |
------------------------------------------------------------------------------------------------------------------
|     Trainer                              MicroAccuracy  MacroAccuracy  Duration #Iteration                     |
|0    DNN + ResNet50                              0,4610         0,4624     228,2          0                     |
------------------------------------------------------------------------------------------------------------------

**Expected behavior**
I expected about 100 epochs and accuracy over 0,8 (%80)

**Screenshots**
![image](https://user-images.githubusercontent.com/48297653/173770637-6b78de58-b81e-4a16-bcdf-6d42da0e761d.png)

**Comment by TheAwerx** [comment]
@LittleLittleCloud 150 classes and about 500 images for each class. Image's sizes are 299x299
![report2](https://user-images.githubusercontent.com/48297653/174068633-c0f4f692-fecc-461c-80c4-c250e33a4a00.PNG)
![report1](https://user-images.githubusercontent.com/48297653/174068647-e61a7592-fdab-40de-a0d4-35cc17cc8c37.PNG)

**Comment by LittleLittleCloud** [comment]
@TheAwerx Is the difference obvious among 150 classes? By __obvious __ I mean can easily being distinguished from human eyes. For example, in the following pictures, the first is adana kebap, the second is beyti kababi and their difference is not obvious. 
![image](https://user-images.githubusercontent.com/16876986/174350185-ae1b6486-cac2-4a4e-88a6-65dc78da17c0.png)
![image](https://user-images.githubusercontent.com/16876986/174350271-3b12181b-40a6-4dc1-9df1-93724d9a9afa.png)

In that situation, DNN can still distinguish them by just `remember` them, but in an unseen dataset, just remembering pictures is not enough, it needs to rely on generalized rules, like color, edge and so on. That's why DNN can classify weather because sunny and rainy has different colors, but it's difficult for DNN to classify different kebaps. 

Since I only see one class of your images so I can't guarantee what causes the low validation score, but it might be because the difference among classes is not obvious, and DNN is just remembering all seen pictures and fail to generalize the rule to classify pictures when validation.

In this case, you need to do more feature engineering and help DNN find generalize rule rather than just throwing everything into model builder. Or adding more images in your training set, or both.

**Comment by TheAwerx** [comment]
@LittleLittleCloud I am grateful for your advices. I will try to improve my dataset and try again later. Hopefully this time I can get better results.

**Comment by LittleLittleCloud** [comment]
Cool, since this issue is related to training performance rather than bugs in model builder, I'm going to close this issue. Feel free to re-open it if you still have questions after improving dataset && retraining.
